### PR TITLE
Move testTracer struct in tracingtest package and reuse it

### PR DESCRIPTION
**Pull request by RomanZavodskikh**
This pull requests is split from https://github.com/zalando/skipper/pull/3318 to refactor change of testTracer to make it more reusable.

### CODE REVIEW

File: filters/scheduler/lifo_test.go [hunk 1]
```diff
@@ -508,7 +474,7 @@ func TestLifoErrors(t *testing.T) {
 
 	<-rt.FirstLoad()
 
-	tracer := &testTracer{MockTracer: mocktracer.New()}
+	tracer := tracingtest.NewMockTracer()
```
[the remark this answers is not in the document]
Comment (AlexanderYastrebov) [reply filters/scheduler/lifo_test.go:477]: E.g. this https://github.com/zalando/skipper/blob/f6f9efb2c7ff93ff940b6f5d60fda3e89e89cfec/filters/auth/jwt_metrics_test.go#L307-L308

File: filters/scheduler/fifo_test.go [hunk 2]
```diff
@@ -296,10 +295,8 @@ func TestFifoWithBody(t *testing.T) {
 			rt := routing.New(ro)
 			defer rt.Close()
 			<-rt.FirstLoad()
-			tracer := &testTracer{MockTracer: mocktracer.New()}
```
[the remark this answers is not in the document]
Comment (RomanZavodskikh) [reply filters/scheduler/fifo_test.go:299]: Not, it was used only in lifo_test.go.

File: filters/scheduler/lifo_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RomanZavodskikh) [reply filters/scheduler/lifo_test.go]: I have updated.

File: tracing/tracingtest/mocktracer.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/mocktracer"
+)
+
+type MockTracer struct {
+	mockTracer *mocktracer.MockTracer
+	spans      atomic.Int32
+}
+
+func NewMockTracer() *MockTracer {
```
[the remark this answers is not in the document]
Comment (RomanZavodskikh) [reply tracing/tracingtest/mocktracer.go]: Renamed.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Review by AlexanderYastrebov — APPROVED** [review]

**Comment by RomanZavodskikh** [comment]
:+1:

File: tracing/tracingtest/mocktracer.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		}
+		select {
+		case <-retry.C:
+		case <-timeout:
+			return nil
+		}
+	}
+}
+
+func (t *MockTracer) Inject(sm opentracing.SpanContext, format any, carrier any) error {
```
Comment (MustafaSaber) [inline tracing/tracingtest/mocktracer.go:47]: I don't see Inject & Extract functions used in this PR

**Comment by AlexanderYastrebov** [comment]
This implementation is not correct - it returns spans whose Tracer() method returns ot.MockTracer and not Skipper's MockTracer.

I am looking whether we can fix that within https://github.com/zalando/skipper/pull/3375

**Pull request merged**